dali/image/tiff.h

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_IMAGE_TIFF_H_
+#define DALI_IMAGE_TIFF_H_
+
+#include <opencv2/opencv.hpp>
+
+#include <string>
+
+#include <vector>
+
+#include "dali/common.h"
+#include "dali/error_handling.h"
+#include "dali/pipeline/data/backend.h"
+#include "dali/pipeline/data/tensor.h"
+
+namespace dali {
+
+
+/**
+ * Class, that handles byte buffer for tiff image
+ */
+class tiff_buffer {
```

Comment:
I see a bit of inconsistency with class names/function names. Sometimes I see CamelCase, sometimes lower_snake_case. Do we have a predefined way? I have my personal preferences but I would settle for a commonly agreed pattern

Reply:
I think we should follow https://google.github.io/styleguide/cppguide.html#General_Naming_Rules, as we state in the https://github.com/NVIDIA/DALI/blob/master/CONTRIBUTING.md.

Reply:
I guess we all follow Google Style. I decided to make it lower_snake to emphasize, that it's a local class, specialized for TIFF image buffer. Later, when it would happen, that this class would be generalized in some way, it can be renamed